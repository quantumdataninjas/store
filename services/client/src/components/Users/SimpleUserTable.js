import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  Grid,
  Table,
  TableHeaderRow
} from '@devexpress/dx-react-grid-material-ui'
import { Paper } from '@material-ui/core'


class SimpleUserTable extends Component {

  static propTypes = {
    simple_users: PropTypes.array.isRequired
  }

  render() {
    let { simple_users } = this.props
    simple_users.forEach(user => {
      for(let key in user){
        if(typeof(user[key]) === "boolean") {
          user[key] = user[key].toString()
        } else if(user[key] instanceof Date) {
          user[key] = user[key].toLocaleDateString()
        }
      }
    })

    return (
      <Paper>
        <Grid
          rows={simple_users}
          columns={[
            { name: 'id', title: 'ID' },
            { name: 'email', title: 'Email' },
            { name: 'subscribed', title: 'Subscribed' },
            { name: 'signed_up', title: 'Signed Up' },
            { name: 'online', title: 'Online' },
            { name: 'last_signin', title: 'Last Signin' },
            { name: 'last_signout', title: 'Last Signout' },
            { name: 'created_at', title: 'Created At' }
          ]}>
          <h1 className="title is-1" style={{color: 'white'}}>Simple User Table</h1>
          <Table styles="border-right: 1px solid white" />
          <TableHeaderRow />
        </Grid>
      </Paper>
    )
  }
}

export default SimpleUserTable
