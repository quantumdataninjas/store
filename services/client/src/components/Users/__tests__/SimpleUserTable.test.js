import React from 'react'
import { mount, shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Grid } from '@devexpress/dx-react-grid-material-ui'
import SimpleUserTable from '../SimpleUserTable'


const simple_users = [
    {
        "id": "1",
        "email": "user@test.org",
        "subscribed": "true",
        "signed_up": "false",
        "online": "true",
        "last_signin": "8/26/2019",
        "last_signout": "8/26/2019",
        "created_at": "8/26/2019"
    },
    {
        "id": "2",
        "email": "user2@test.org",
        "subscribed": "true",
        "signed_up": "false",
        "online": "true",
        "last_signin": "8/26/2019",
        "last_signout": "8/26/2019",
        "created_at": "8/26/2019"
    },
    {
        "id": "3",
        "email": "test@test.org",
        "subscribed": "true",
        "signed_up": "false",
        "online": "true",
        "last_signin": "8/26/2019",
        "last_signout": "8/26/2019",
        "created_at": "8/26/2019"
    },
]

const cols = [
    { header: "ID", name: "id" },
    { header: "Email", name: "email" },
    { header: "Subscribed", name: "subscribed" },
    { header: "Signed Up", name: "signed_up" },
    { header: "Online", name: "online" },
    { header: "Last Signin", name: "last_signin" },
    { header: "Last Signout", name: "last_signout" },
    { header: "Created At", name: "created_at" },
]

describe('SimpleUserTable Component', () => {

    it('renders empty message as table cell if there is no data', () => {
        const wrapper = mount(<SimpleUserTable simple_users={[]}/>)
        const grid = wrapper.find(Grid);
        expect(grid).toHaveLength(1);
        const table = wrapper.find('table');
        expect(table).toHaveLength(1);
        const thead = table.find('thead');
        expect(thead).toHaveLength(1);
        const headers = thead.find('th');
        expect(headers).toHaveLength(cols.length);
        headers.forEach((th, idx) => {
            expect(th.text()).toEqual(cols[idx].header);
        });
        const tbody = table.find('tbody');
        expect(tbody).toHaveLength(1);
        const row = tbody.find('tr');
        expect(row).toHaveLength(1);
        const cell = row.find('td');
        expect(cell).toHaveLength(1);
        expect(cell.prop('colSpan')).toEqual(cols.length);
        expect(cell.text()).toEqual('No data');
    })

    it('renders', () => {
        //   const wrapper = shallow(<SimpleUserTable simple_users={simple_users}/>)
        const wrapper = mount(<SimpleUserTable simple_users={simple_users}/>)
        const table = wrapper.find('table')
        expect(table).toHaveLength(1)
        const thead = table.find('thead')
        expect(thead).toHaveLength(1)
        const headers = thead.find('th')
        expect(headers).toHaveLength(cols.length)
        headers.forEach((th, i) => {
            expect(th.text()).toEqual(cols[i].header)
        })
        const tbody = table.find('tbody')
        expect(tbody).toHaveLength(1)
        const rows = tbody.find('tr')
        expect(rows).toHaveLength(simple_users.length)
        rows.forEach((tr, i) => {
            const cells = tr.find('td')
            expect(cells).toHaveLength(cols.length)
            cells.forEach((cell, j) => {
                expect(cell.text()).toEqual(simple_users[i][cols[j].name])
            })
        })
    })

    it('renders a snapshot', () => {
        const tree = renderer.create(<SimpleUserTable simple_users={simple_users} />)
        expect(tree).toMatchSnapshot()
    })
})