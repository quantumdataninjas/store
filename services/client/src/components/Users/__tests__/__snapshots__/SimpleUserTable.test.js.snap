// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleUserTable Component renders a snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <h1
    className="title is-1"
    style={
      Object {
        "color": "white",
      }
    }
  >
    Simple User Table
  </h1>
  <div
    className="RootBase-root-28"
  >
    <div
      className="TableContainer-root-29"
    >
      <div>
        <table
          className="MuiTable-root Table-table-30"
          style={
            Object {
              "minWidth": "960px",
            }
          }
        >
          <colgroup>
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
            <col />
          </colgroup>
          <thead
            className="MuiTableHead-root"
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head"
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      ID
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Email
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Subscribed
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Signed Up
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Online
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Last Signin
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Last Signout
                    </span>
                  </div>
                </div>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root TableHeaderCell-cell-42 TableHeaderCell-cellNoWrap-52 MuiTableCell-head"
                colSpan={1}
                scope="col"
                style={null}
              >
                <div
                  className="TableHeaderCell-container-51"
                >
                  <div
                    className="Content-content-64"
                  >
                    <span
                      className="Title-title-67"
                    >
                      Created At
                    </span>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
          >
            <tr
              className="MuiTableRow-root"
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                1
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                user@test.org
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                false
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
            </tr>
            <tr
              className="MuiTableRow-root"
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                2
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                user2@test.org
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                false
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
            </tr>
            <tr
              className="MuiTableRow-root"
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                3
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                test@test.org
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                false
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                true
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
              <td
                aria-sort={null}
                className="MuiTableCell-root TableCell-cell-72 TableCell-cellNoWrap-76 MuiTableCell-body"
                colSpan={1}
              >
                8/26/2019
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
